parity-clib/examples/cpp/main.cpp

```diff
@@ -49,58 +48,38 @@ const std::vector<std::string> ws_subscriptions {
 	"{\"method\":\"eth_subscribe\",\"params\":[\"newHeads\"],\"id\":1,\"jsonrpc\":\"2.0\"}"
 };
 
+std::atomic<int> callback_counter;
+
 // callback that gets invoked upon an event
 void callback(void* user_data, const char* response, size_t _len) {
 	Callback* cb = static_cast<Callback*>(user_data);
 	if (cb->type == CALLBACK_RPC) {
-		cb->counter -= 1;
+		callback_counter -= 1;
 	} else if (cb->type == CALLBACK_WS) {
 		std::regex is_subscription ("\\{\"jsonrpc\":\"2.0\",\"result\":\"0[xX][a-fA-F0-9]{16}\",\"id\":1\\}");
 		if (std::regex_match(response, is_subscription) == true) {
-			cb->counter -= 1;
+			callback_counter -= 1;
 		}
 	}
 }
 
 int main() {
-	// run full-client
-	{
-		std::vector<const char*> config = {"--no-ipc" , "--jsonrpc-apis=all", "--chain", "kovan"};
-		void* parity = parity_run(config);
-		if (parity_rpc_queries(parity)) {
-			printf("rpc_queries failed\r\n");
-			return 1;
-		}
-
-		if (parity_subscribe_to_websocket(parity)) {
-			printf("ws_queries failed\r\n");
-			return 1;
-		}
-
-		if (parity != nullptr) {
-			parity_destroy(parity);
-		}
+	std::vector<const char*> config = {"--no-ipc" , "--jsonrpc-apis=all", "--chain", "kovan"};
+	void* parity = parity_run(config);
```

Comment:
```suggestion
	void *parity = parity_run(config);
```

would be a better match to how this is actually parsed by the C++ compiler.  However, the use of a raw pointer ― _especially_ a `void *` ― is an anti-pattern in C++.  It would be much better to use a proper C++ class with a destructor, following RAII.  Otherwise, it is highly likely that resources will be leaked.  This is especially important in C++, as using exceptions in C++ is considered ideomatic.  In this case, the resources will be destroyed when `main()` returns and the program exits, but that is still a bad example.